hack/e2e/test.sh

```diff
@@ -195,7 +194,38 @@ kubectl -n team-1 wait certificate app-0 --for=condition=Ready
 # Parse logs as JSON using jq to ensure logs are all JSON formatted.
 # Disable pipefail to prevent SIGPIPE (141) errors from tee
 # See https://unix.stackexchange.com/questions/274120/pipe-fail-141-when-piping-output-into-tee-why
+set +o pipefail
 kubectl logs deployments/venafi-kubernetes-agent \
         --follow \
         --namespace venafi \
     | timeout 60 jq 'if .msg | test("Data sent successfully") then . | halt_error(0) end'
+set -o pipefail
```

Comment:
I'd accidentally removed this pipefail trick in https://github.com/jetstack/jetstack-secure/pull/596
and it resulted in the test always exiting with a 141 status.

I've restored it here so that the new shell lines can run after the "Data sent successfully" message.

Reply:
Thanks for fixing this! I was seeing this 141 exit code without really paying attention to it since it said "Data sent successfully" 😅